src/Simplex/Chat.hs

```diff
@@ -6014,13 +6118,29 @@ sameGroupProfileInfo p p' = p {groupPreferences = Nothing} == p' {groupPreferenc
 
 createInternalChatItem :: forall c d m. (ChatTypeI c, MsgDirectionI d, ChatMonad m) => User -> ChatDirection c d -> CIContent d -> Maybe UTCTime -> m ()
 createInternalChatItem user cd content itemTs_ = do
+  ci <- createInternalChatItem_ user cd content itemTs_
+  toView $ CRNewChatItem user (AChatItem (chatTypeI @c) (msgDirection @d) (toChatInfo cd) ci)
+
+createInternalChatItem_ :: (MsgDirectionI d, ChatMonad m) => User -> ChatDirection c d -> CIContent d -> Maybe UTCTime -> m (ChatItem c d)
+createInternalChatItem_ user cd content itemTs_ = do
   createdAt <- liftIO getCurrentTime
   let itemTs = fromMaybe createdAt itemTs_
   ciId <- withStore' $ \db -> do
     when (ciRequiresAttention content) $ updateChatTs db user cd createdAt
     createNewChatItemNoMsg db user cd content itemTs createdAt
-  ci <- liftIO $ mkChatItem cd ciId content Nothing Nothing Nothing Nothing False itemTs Nothing createdAt
-  toView $ CRNewChatItem user (AChatItem (chatTypeI @c) (msgDirection @d) (toChatInfo cd) ci)
+  liftIO $ mkChatItem cd ciId content Nothing Nothing Nothing Nothing False itemTs Nothing createdAt
+
+createLocalChatItem :: (MsgDirectionI d, ChatMonad m) => User -> ChatDirection 'CTLocal d -> CIContent d -> Maybe UTCTime -> m (ChatItem 'CTLocal d)
+createLocalChatItem user cd content itemTs_ = do
+  createdAt <- liftIO getCurrentTime
+  let itemTs = fromMaybe createdAt itemTs_
+  gVar <- asks random
+  ciId <- withStore $ \db -> do
+    when (ciRequiresAttention content) . liftIO $ updateChatTs db user cd createdAt
+    createWithRandomId gVar $ \sharedMsgId ->
+      let smi_ = Just (SharedMsgId sharedMsgId)
+       in createNewChatItem_ db user cd Nothing smi_ content (Nothing, Nothing, Nothing, Nothing, Nothing) Nothing False itemTs Nothing createdAt
+  liftIO $ mkChatItem cd ciId content Nothing Nothing Nothing Nothing False itemTs Nothing createdAt
```

Comment:
how does it work when chat item has file if it always passes Nothing to Maybe CIFile? it seems files will not be displayed in UI, probably need to add tests where items with files are created and chat is read. see tests by search `"messages with files"`

Reply:
ciFile_ gets attached later with a record update

Reply:
ok, I see now. In case of local file it may be better to save both chat item and file in a single transaction, it seems there's no work happening in between (like preparation of file invitation / requesting agent to send xftp file / preparation of sharedMsgId).

Reply:
You wouldn't have to go around this function then and make a record update. This may become a trap if mkChatItem gets some additional logic later. These record updates (or lack of them) after creating a record in database were an often source of bugs at some point.. we try to avoid it now